cmd/kcp/kcp.go

```diff
@@ -219,7 +223,38 @@ func main() {
 				prepared := server.PrepareRun()
 
 				return prepared.Run(ctx.Done())
-			})
+			}
+
+			if len(etcdClientInfo.Endpoints) == 0 {
+				// No etcd servers specified so create one in-process:
+				return s.Run(runFunc)
+			} else {
```

Comment:
style nit: Since the previous `if` returns, you can skip the `else` here and unindent the rest of this.